android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/LanguageResource.java

```diff
@@ -45,5 +64,71 @@ public int hashCode() {
 
   public abstract Bundle buildDownloadBundle();
 
-  //public abstract boolean equals();
+  /**
+   * Constructor using JSON Objects from installed language resource list
+   * @param installedObj
+   */
+  public LanguageResource(JSONObject installedObj) {
+    this.fromJSON(installedObj);
+  }
```

Comment:
We don't need this because the subclasses are responsible for calling their `super.fromJSON()` methods at the appropriate time. Implementation of `fromJSON()` here is correct, but responsibility for instantiation belongs in the instantiated classes.